scripts/deploy-beta-apm.js

```diff
@@ -1,63 +1,104 @@
 const namehash = require('eth-ens-namehash').hash
 const keccak256 = require('js-sha3').keccak_256
 
-const APMRegistryFactory = artifacts.require('APMRegistryFactory')
-const DAOFactory = artifacts.require('DAOFactory')
-const ENS = artifacts.require('ENS')
+const deployENS = require('./deploy-beta-ens')
 
-const owner = process.env.OWNER || '0x4cb3fd420555a09ba98845f0b816e45cfb230983'
-const ens = process.env.ENS || '0xfbae32d1cde62858bc45f51efc8cc4fa1415447e'
+const globalArtifacts = this.artifacts // Not injected unless called directly via truffle
 
-const tld = namehash('eth')
-const label = '0x'+keccak256('aragonpm')
-
-const getContract = name => artifacts.require(name)
+const defaultOwner = process.env.OWNER || '0x4cb3fd420555a09ba98845f0b816e45cfb230983'
+const defaultENS = process.env.ENS
 
 const baseInitArguments = {
   Kernel: [ true ] // petrify
 }
 
-const deployBases = async baseNames => {
-  const baseContracts = await Promise.all(baseNames.map(c => getContract(c).new(...(baseInitArguments[c] || []))))
-  return baseContracts.map(c => c.address)
+const deployBases = async baseContracts => {
+  const deployedContracts = await Promise.all(baseContracts.map(c => c.new(...(baseInitArguments[c.contractName] || []))))
+  return deployedContracts.map(c => c.address)
 }
 
-module.exports = async callback => {
-  console.log('Deploying APM')
-  console.log('Owner:', owner)
-  console.log('ENS:', ens)
-  console.log('TLD:', tld)
-  console.log('Label:', label)
+module.exports = async (truffleExecCallback, { artifacts = globalArtifacts, ens = defaultENS, owner = defaultOwner, verbose = true } = {}) => {
+  const log = (...args) => {
+    if (verbose) { console.log(...args) }
+  }
+
+  const ACL = artifacts.require('ACL')
+  const Kernel = artifacts.require('Kernel')
+  const APMRegistry = artifacts.require('APMRegistry')
+  const Repo = artifacts.require('Repo')
+  const ENSSubdomainRegistrar = artifacts.require('ENSSubdomainRegistrar')
+
+  const APMRegistryFactory = artifacts.require('APMRegistryFactory')
+  const DAOFactory = artifacts.require('DAOFactory')
+  const ENS = artifacts.require('ENS')
+
+  const tldName = 'eth'
+  const labelName = 'aragonpm'
+  const tldHash = namehash(tldName)
+  const labelHash = '0x'+keccak256(labelName)
 
-  console.log('=========')
-  console.log('deploying APM bases')
-  const apmBases = await deployBases(['APMRegistry', 'Repo', 'ENSSubdomainRegistrar'])
-  console.log('deployed APM bases', apmBases)
+  log('Deploying APM...')
+  log('Owner:', owner)
 
-  console.log('deploying DAO bases')
-  const daoBases = await deployBases(['Kernel', 'ACL'])
-  console.log('deployed DAO bases', daoBases)
+  if (!ens) {
+    log('=========')
+    log('Missing ENS! Deploying a custom ENS...')
+    ens = (await deployENS(null, { artifacts, owner, verbose: false })).ens
```

Comment:
Shouldn't the `verbose` param passed in here instead of `false`?

Reply:
I wanted to silence all the logging in `deployENS()`, so that this script can control what gets logged.